add-on/panels/test/desktopViews_test.js

```diff
@@ -0,0 +1,352 @@
+    });
+
+    it("should dispatch an EmailRoomUrl with no description" +
+       " for rooms without context when the email button is pressed",
+      function() {
+        var emailBtn = view.getDOMNode();
+
+        React.addons.TestUtils.Simulate.click(emailBtn);
+
+        sinon.assert.calledOnce(dispatcher.dispatch);
+        sinon.assert.calledWith(dispatcher.dispatch,
+          new sharedActions.EmailRoomUrl({
+            roomUrl: "http://invalid",
+            roomDescription: undefined,
+            from: "conversation"
+          }));
+      });
+
+    it("should dispatch an EmailRoomUrl with a domain name description for rooms with context",
+      function() {
+        view = mountTestComponent({
+          roomData: {
+            roomUrl: "http://invalid",
+            roomContextUrls: [{ location: "http://www.mozilla.com/" }]
+          }
+        });
+
+        var emailBtn = view.getDOMNode();
+
+        React.addons.TestUtils.Simulate.click(emailBtn);
+
+        sinon.assert.calledOnce(dispatcher.dispatch);
+        sinon.assert.calledWith(dispatcher.dispatch,
+          new sharedActions.EmailRoomUrl({
+            roomUrl: "http://invalid",
+            roomDescription: "www.mozilla.com",
+            from: "conversation"
+          }));
+      });
+
+    it("should invoke callback if defined", function() {
+      var callback = sinon.stub();
+      view = mountTestComponent({
+        callback: callback
+      });
+      var emailBtn = view.getDOMNode();
+      React.addons.TestUtils.Simulate.click(emailBtn);
+
+      sinon.assert.calledOnce(callback);
+    });
+  });
+
+  describe("FacebookShareButton", function() {
+    var view;
+
+    function mountTestComponent(props) {
+      props = _.extend({
+        dispatcher: dispatcher,
+        from: "conversation",
+        roomData: {
+          roomUrl: "http://invalid",
+          roomContextUrls: []
+        }
+      }, props || {});
+      return TestUtils.renderIntoDocument(
+        React.createElement(sharedDesktopViews.FacebookShareButton, props));
+    }
+
+    it("should dispatch a FacebookShareRoomUrl action when the facebook button is clicked",
+      function() {
+        view = mountTestComponent();
+
+        var facebookBtn = view.getDOMNode();
+
+        React.addons.TestUtils.Simulate.click(facebookBtn);
+
+        sinon.assert.calledOnce(dispatcher.dispatch);
+        sinon.assert.calledWith(dispatcher.dispatch,
+          new sharedActions.FacebookShareRoomUrl({
+            from: "conversation",
+            roomUrl: "http://invalid"
+          }));
+    });
+
+    it("should invoke callback if defined", function() {
+      var callback = sinon.stub();
+      view = mountTestComponent({
+        callback: callback
+      });
+      var facebookBtn = view.getDOMNode();
+      React.addons.TestUtils.Simulate.click(facebookBtn);
+
+      sinon.assert.calledOnce(callback);
+    });
+  });
+
+  describe("SharePanelView", function() {
```

Comment:
Please can you also add a couple of tests to check that it isn't displayed when the prop `show` is false, nor when `this.props.roomData.roomUrl` is false.